frontend/packages/shipwright-plugin/src/components/build-tabsection/BuildOverview.tsx

```diff
@@ -0,0 +1,122 @@
+import { OverviewItem } from '@console/shared';
+import { BUILDRUN_TO_RESOURCE_MAP_LABEL } from '../../const';
+import { BuildModel } from '../../models';
+import { Build, BuildRun } from '../../types';
+import BuildRunItem from './BuildRunItem';
+import StartBuildButton from './StartBuildButton';
+import TriggerLastBuildButton from './TriggerLastBuildButton';
+
+const MAX_VISIBLE = 3;
+
+type BuildsOverviewProps = {
+  item: OverviewItem & {
+    builds?: Build[];
+    buildRuns?: BuildRun[];
+  };
+};
+
+const BuildsOverview: React.FC<BuildsOverviewProps> = ({ item: { builds, buildRuns, obj } }) => {
+  const { t } = useTranslation();
+  const buildRunsforResource = _.filter(buildRuns, (buildRun) => {
+    return (
+      buildRun.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL] ===
+      obj.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL]
+    );
+  });
+
+  if (!builds || !builds.length) {
+    return null;
+  }
+
+  return (
+    <>
+      <SidebarSectionHeading text={t('shipwright-plugin~BuildRuns')}>
+        {buildRunsforResource.length > MAX_VISIBLE && (
+          <Link
+            className="sidebar__section-view-all"
+            to={`/k8s/ns/${obj.metadata?.namespace}/shipwright.io~v1alpha1~BuildRun?labels=${BUILDRUN_TO_RESOURCE_MAP_LABEL}=${obj.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL]}`}
+          >
+            {t('shipwright-plugin~View all {{buildRunsLength}}', {
+              buildRunsLength: buildRunsforResource.length,
+            })}
+          </Link>
+        )}
+      </SidebarSectionHeading>
+      {builds.map((build) => {
+        const buildRunsforBuild = buildRuns.filter(
+          (buildRun) =>
+            buildRun.spec.buildRef?.name === build.metadata.name &&
+            buildRun.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL] ===
+              obj.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL],
+        );
+        return (
+          <ul className="list-group pf-u-mb-xl">
+            <li className="list-group-item">
+              <Flex justifyContent={{ default: 'justifyContentSpaceBetween' }}>
+                <FlexItem>
+                  <ResourceLink
+                    inline
+                    kind={referenceForModel(BuildModel)}
+                    name={build.metadata.name}
+                    namespace={build.metadata.namespace}
+                  />
+                </FlexItem>
+                <FlexItem>
+                  {buildRunsforBuild.length > MAX_VISIBLE && (
+                    <Link
+                      className="sidebar__section-view-all"
+                      to={`${resourcePath(
+                        referenceForModel(BuildModel),
+                        build.metadata.name,
+                        build.metadata.namespace,
+                      )}/buildruns?labels=${BUILDRUN_TO_RESOURCE_MAP_LABEL}=${
+                        obj.metadata?.labels?.[BUILDRUN_TO_RESOURCE_MAP_LABEL]
+                      }`}
```

Comment:
Also if a label could not contain JavaScript like an annotation, please let us start to encode URLs that we build dynamically:

```suggestion
                      )}/buildruns?labels=${BUILDRUN_TO_RESOURCE_MAP_LABEL}=${
                        encodeURIComponent(resourceLabel)
                      }`}
```